libs/networking/sources/Snapshot.cpp

```diff
@@ -28,45 +30,31 @@ using ComponentId = uint8_t;
 using UpdateType = bool;
 
 static void diffAdd(
-    std::vector<std::byte> &diff,
+    engine::Serializer &diff,
     std::vector<ComponentData>::const_iterator const &it
 )
 {
-    size_t offset = diff.size();
-    diff.resize(
-        offset + sizeof(EntityNumber) + sizeof(ComponentId) + sizeof(UpdateType)
-            + it->data.size(),
-        std::byte(0)
-    );
-    std::memcpy(&diff[offset], &it->entity, sizeof(EntityNumber));
-    offset += sizeof(EntityNumber);
-    std::memcpy(&diff[offset], &it->componentId, sizeof(ComponentId));
-    offset += sizeof(ComponentId);
-    diff[offset] = std::byte(0x01);
-    offset += sizeof(UpdateType);
-    std::memcpy(&diff[offset], it->data.data(), it->data.size());
+    diff.serializeTrivial(EntityNumber(it->entity));
+    diff.serializeTrivial(ComponentId(it->componentId));
+    diff.serializeTrivial(UpdateType(0x01));
```

Comment:
pour les updateType dans le fichier :
Clang-Tidy: Converting integer literal to bool, use bool literal instead

Reply:
Le protocole le spécifie comme ça donc je garde comme ça